android/src/main/java/com/oney/WebRTCModule/PermissionUtils.java

```diff
@@ -0,0 +1,133 @@
+
+        Intent permissionsIntent = new Intent(context, PermissionHelperActivity.class);
+        permissionsIntent.putExtra(RESULT_RECEIVER, resultReceiver);
+        permissionsIntent.putExtra(PERMISSION, permission);
+        permissionsIntent.putExtra(REQUEST_CODE, requestCode);
+        permissionsIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+        context.startActivity(permissionsIntent);
+    }
+
+    /**
+     * Helper activity for requesting permissions. Android only allows requesting permissions from
+     * an activity and the result is reported in the <tt>onRequestPermissionsResult</tt> method.
+     * Since this package is a library we create an auxiliary activity and communicate back the
+     * results using a <tt>ResultReceiver</tt>.
+     */
+    public static class PermissionHelperActivity extends Activity {
+        int requestCode;
+        ResultReceiver resultReceiver;
+        String permission;
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void onRequestPermissionsResult(int requestCode,
+                                               String permissions[],
+                                               int[] grantResults) {
+            Bundle resultData = new Bundle();
+            resultData.putString(PERMISSION, permissions[0]);
+            resultData.putInt(GRANT_RESULT, grantResults[0]);
+            resultReceiver.send(requestCode, resultData);
+            finish();
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        protected void onStart() {
+            super.onStart();
+
+            resultReceiver = this.getIntent().getParcelableExtra(RESULT_RECEIVER);
+            permission = this.getIntent().getStringExtra(PERMISSION);
+            requestCode = this.getIntent().getIntExtra(REQUEST_CODE, 0);
+
+            // Here we test for the target SDK version with which *the app* was compiled. If we use
+            // Build.VERSION.SDK_INT that would give us the API version of the device itself, not
+            // the version the app was compiled for. When compiled for API level < 23 we must still
+            // use old permissions model, regardless of the Android version on the device.
+            final int targetSdkVersion = getApplicationInfo().targetSdkVersion;
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
+                    && targetSdkVersion >= Build.VERSION_CODES.M) {
+                ActivityCompat.requestPermissions(
+                    this,new String[]{permission}, requestCode);
+            } else {
+                // In the old permissions model the permissions are granted at install time, we can
+                // check for them directly.
+                Bundle resultData = new Bundle();
+                resultData.putString(PERMISSION, permission);
+                resultData.putInt(
+                    GRANT_RESULT, ContextCompat.checkSelfPermission(this, permission));
```

Comment:
https://developer.android.com/training/permissions/requesting.html says that `ContextCompat.checkSelfPermission` should be called in both branches of the above if in order to avoid a call to `ActivityCompat.requestPermissions` when the permission is granted already.